components/match2/commons/opponent.lua

```diff
@@ -343,4 +343,85 @@ function Opponent.fromMatch2Record(record)
 	end
 end
 
+--[[
+Reads an opponent struct and builds a storage struct for standing/prizePool from it
+]]
+function Opponent.toLpdbStruct(opponent, resolveTeamToPageName)
+	-- this already handles literal and team opponents
+	-- (modulo resolveTeamToPageName if needed)
+	local storageStruct = {
+		opponentname = opponent.name,
+		opponenttemplate = opponent.template,
+		opponenttype = opponent.type,
+	}
+
+	if opponent.type == Opponent.team then
+		if resolveTeamToPageName then
+			storageStruct.opponentname = Opponent.teamTemplateResolveRedirect(opponent.template)
+		end
+	elseif Opponent.typeIsParty(opponent.type) then
+		local players = {}
+		for playerIndex, player in ipairs(opponent.players) do
+			players['p' .. playerIndex .. 'dn'] = player.displayName
+			players['p' .. playerIndex .. 'flag'] = player.flag
+			players['p' .. playerIndex] = player.pageName
+			players['p' .. playerIndex .. 'team'] = resolveTeamToPageName
+				and Opponent.teamTemplateResolveRedirect(player.team)
+				or player.team
+			players['p' .. playerIndex .. 'template'] = resolveTeamToPageName
+				and player.team or nil
+		end
+		storageStruct.opponentplayers = players
+	end
+
+	return storageStruct
+end
+
+-- function to resolve teamTemplate to pagename and resolve redirect on it
+-- needed for wikis that store redirected
+-- also works as a workaround if there are still duplicate teamTemplates on a wiki
+-- or if the teamplate name does not fit the pagename while the pagename is already another teamTemplate
+-- (which btw. can not be fixed in several cases due to the lock due to std logos)
+function Opponent.teamTemplateResolveRedirect(template)
+	if String.isEmpy(template) then
+		return nil
+	end
+
+	if mw.ext.TeamTemplate.teamexists(template) then
+		local page = mw.ext.TeamTemplate.raw(template).page
+		return mw.ext.TeamLiquidIntegration.resolve_redirect(page)
+	end
+
+	return template
+end
+
+--[[
+Reads a standing/prizePool storage struct and builds an opponent struct from it
+]]
+function Opponent.fromLpdbStruct(storageStruct)
+	local partySize = Opponent.partySize(storageStruct.type)
+	if partySize then
+		local players = storageStruct.opponentplayers
+		local function playerFromLpdbStruct(playerIndex)
+			return {
+				displayName = players['p' .. playerIndex .. 'dn'],
+				flag = Flags.CountryName(players['p' .. playerIndex .. 'flag']),
+				pageName = players['p' .. playerIndex],
+				team = players['p' .. playerIndex .. 'team'],
+			}
+		end
+		local opponent = {
+			players = Array.map(Array.range(1, partySize), playerFromLpdbStruct),
+			type = storageStruct.type,
+		}
+		return opponent
+	else -- literal and team opponents can be handled the same way
```

Comment:
Prefer have them separated in order to keep the struct clean (`template` for team, `name` for literal)

Reply:
name is also a thing for team opponents, so it is only about the template, but i will duplicate the lines as wanted